python/ray/actor_mesh.py

```diff
@@ -0,0 +1,194 @@
+import copy
+import math
+import os
+import random
+from typing import Any, Callable, Concatenate, Dict, Generic, List, Optional, Tuple, TypeVar, ParamSpec, Union
+from typing import overload
+
+import ray
+from ray.actor import ActorHandle, ActorProxy
+
+T = TypeVar("T", covariant=True)
+
+P = ParamSpec("P")
+R = TypeVar("R")
+
+class ActorMeshProxy:
+    def __init__(self, actor_mesh: "ActorMesh") -> None:
+        self.actor_mesh = actor_mesh
+
+    def __getattr__(self, k):
+        return ActorMethodProxy(self.actor_mesh, k)
+    
+class ActorMethod(Generic[P, R]):
+
+    def __init__(self, method):
+        self.method = method
+
+    def all(self, *args: P.args, **kwargs: P.kwargs) -> List["ray.ObjectRef[R]"]:
+        "Call all the actors in the mesh with the same arguments."
+        pass
+
+    def choose(self, *args: P.args, **kwargs: P.kwargs) -> "ray.ObjectRef[R]":
+        "Call one of the actors in the mesh."
+        pass
+        
+    def shard(self, *args: P.args, **kwargs: P.kwargs) -> R:
+        "Call all the actors in the mesh, shard the argument(s) among them."
+        pass
+    
+class ActorMethodProxy:
+
+    def __init__(self, actor_mesh: "ActorMesh", actor_method: str) -> None:
+        self.actor_mesh = actor_mesh
+        self.actor_method = actor_method
+
+    def all(self, *args, **kwargs):
+        results = []
+        for actor in self.actor_mesh.actors:
+            method = getattr(actor, self.actor_method)
+            results.append(method.remote(*args, **kwargs))
+        return results
+    
+    def choose(self, *args, **kwargs):
```

Comment:
Should it be push-based or pull-based? Or should we support both? If the use case involves data, pull-based might be a better option because of long-tail tasks.